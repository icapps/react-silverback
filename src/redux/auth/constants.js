export default {
  LOGIN_USER: 'LOGIN_USER',
  LOGIN_USER_PENDING: 'LOGIN_USER_PENDING',
  LOGIN_USER_REJECTED: 'LOGIN_USER_REJECTED',
  LOGIN_USER_FULFILLED: 'LOGIN_USER_FULFILLED',
  LOGOUT_USER: 'LOGOUT_USER',
};
