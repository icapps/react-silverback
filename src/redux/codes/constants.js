export default {
  GET_LANGUAGE_CODES: 'GET_LANGUAGE_CODES',
  GET_LANGUAGE_CODES_PENDING: 'GET_LANGUAGE_CODES_PENDING',
  GET_LANGUAGE_CODES_FULFILLED: 'GET_LANGUAGE_CODES_FULFILLED',
  GET_LANGUAGE_CODES_REJECTED: 'GET_LANGUAGE_CODES_REJECTED',

  GET_LANGUAGE_CODE_BY_ID: 'GET_LANGUAGE_CODE_BY_ID',
  GET_LANGUAGE_CODE_BY_ID_PENDING: 'GET_LANGUAGE_CODE_BY_ID_PENDING',
  GET_LANGUAGE_CODE_BY_ID_FULFILLED: 'GET_LANGUAGE_CODE_BY_ID_FULFILLED',
  GET_LANGUAGE_CODE_BY_ID_REJECTED: 'GET_LANGUAGE_CODE_BY_ID_REJECTED',

  CREATE_LANGUAGE_CODE: 'CREATE_LANGUAGE_CODE',
  CREATE_LANGUAGE_CODE_PENDING: 'CREATE_LANGUAGE_CODE_PENDING',
  CREATE_LANGUAGE_CODE_FULFILLED: 'CREATE_LANGUAGE_CODE_FULFILLED',
  CREATE_LANGUAGE_CODE_REJECTED: 'CREATE_LANGUAGE_CODE_REJECTED',

  DEPRECATE_LANGUAGE_CODE: 'DEPRECATE_LANGUAGE_CODE',
  DEPRECATE_LANGUAGE_CODE_PENDING: 'DEPRECATE_LANGUAGE_CODE_PENDING',
  DEPRECATE_LANGUAGE_CODE_FULFILLED: 'DEPRECATE_LANGUAGE_CODE_FULFILLED',
  DEPRECATE_LANGUAGE_CODE_REJECTED: 'DEPRECATE_LANGUAGE_CODE_REJECTED',

  UNDEPRECATE_LANGUAGE_CODE: 'UNDEPRECATE_LANGUAGE_CODE',
  UNDEPRECATE_LANGUAGE_CODE_PENDING: 'UNDEPRECATE_LANGUAGE_CODE_PENDING',
  UNDEPRECATE_LANGUAGE_CODE_FULFILLED: 'UNDEPRECATE_LANGUAGE_CODE_FULFILLED',
  UNDEPRECATE_LANGUAGE_CODE_REJECTED: 'UNDEPRECATE_LANGUAGE_CODE_REJECTED',

  GET_STATUS_CODES: 'GET_STATUS_CODES',
  GET_STATUS_CODES_PENDING: 'GET_STATUS_CODES_PENDING',
  GET_STATUS_CODES_REJECTED: 'GET_STATUS_CODES_REJECTED',
  GET_STATUS_CODES_FULFILLED: 'GET_STATUS_CODES_FULFILLED',

  SET_CODES_SORT: 'SET_CODES_SORT',
};
