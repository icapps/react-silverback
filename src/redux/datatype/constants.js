export default {
  GET_DATA_TYPE: 'GET_DATA_TYPE',
  GET_DATA_TYPE_PENDING: 'GET_DATA_TYPE_PENDING',
  GET_DATA_TYPE_REJECTED: 'GET_DATA_TYPE_REJECTED',
  GET_DATA_TYPE_FULFILLED: 'GET_DATA_TYPE_FULFILLED',
};
