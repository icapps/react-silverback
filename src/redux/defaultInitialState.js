const defaultInitialState = {
  isPending: false,
  isError: false,
  errorMessage: '',
};

export default defaultInitialState;
