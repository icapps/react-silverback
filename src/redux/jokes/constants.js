export default {
  GET_JOKE           : 'GET_JOKE',
  GET_JOKE_REQUEST   : 'GET_JOKE_REQUEST',
  GET_JOKE_FULFILLED : 'GET_JOKE_FULFILLED',
  GET_JOKE_REJECTED  : 'GET_JOKE_REJECTED',
};
