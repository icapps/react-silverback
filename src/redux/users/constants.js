export default {
  GET_USERS: 'GET_USERS',
  GET_USERS_PENDING: 'GET_USERS_PENDING',
  GET_USERS_REJECTED: 'GET_USERS_REJECTED',
  GET_USERS_FULFILLED: 'GET_USERS_FULFILLED',

  GET_USERS_BY_ID: 'GET_USERS_BY_ID',
  GET_USERS_BY_ID_PENDING: 'GET_USERS_BY_ID_PENDING',
  GET_USERS_BY_ID_REJECTED: 'GET_USERS_BY_ID_REJECTED',
  GET_USERS_BY_ID_FULFILLED: 'GET_USERS_BY_ID_FULFILLED',

  CREATE_USER: 'CREATE_USER',
  CREATE_USER_PENDING: 'CREATE_USER_PENDING',
  CREATE_USER_REJECTED: 'CREATE_USER_REJECTED',
  CREATE_USER_FULFILLED: 'CREATE_USER_FULFILLED',

  UPDATE_USER: 'UPDATE_USER',
  UPDATE_USER_PENDING: 'UPDATE_USER_PENDING',
  UPDATE_USER_REJECTED: 'UPDATE_USER_REJECTED',
  UPDATE_USER_FULFILLED: 'UPDATE_USER_FULFILLED',

  REMOVE_USER: 'REMOVE_USER',
  REMOVE_USER_PENDING: 'REMOVE_USER_PENDING',
  REMOVE_USER_REJECTED: 'REMOVE_USER_REJECTED',
  REMOVE_USER_FULFILLED: 'REMOVE_USER_FULFILLED',

  GET_USER_ROLES: 'GET_USER_ROLES',
  GET_USER_ROLES_PENDING: 'GET_USER_ROLES_PENDING',
  GET_USER_ROLES_REJECTED: 'GET_USER_ROLES_REJECTED',
  GET_USER_ROLES_FULFILLED: 'GET_USER_ROLES_FULFILLED',

  SET_USERS_SORT: 'SET_USERS_SORT',
};
