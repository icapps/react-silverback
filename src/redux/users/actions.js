import constants from './constants';
import { userService } from '../../services';

export const getUsers = (page, limit, sortField, sortOrder) => {
  return {
    type: constants.GET_USERS,
    payload: userService.get(page, limit, sortField, sortOrder),
  };
};

export const getUsersById = id => {
  return {
    type: constants.GET_USERS_BY_ID,
    payload: userService.getById(id),
  };
};

export const updateUser = (id, user) => {
  return {
    type: constants.UPDATE_USER,
    payload: userService.update(id, user),
  };
};
