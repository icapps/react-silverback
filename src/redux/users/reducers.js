import constants from './constants';
import defaultInitialState from '../defaultInitialState';

const initialState = {
  userList: [],
  user: null,
  usersCount: 0,
  ...defaultInitialState,
};

const users = (state = initialState, action = {}) => {
  const { error, payload } = action;

  switch (action.type) {
    case constants.GET_USERS_FULFILLED:
      return {
        ...state,
        userList: payload.data,
        usersCount: payload.meta.totalCount,
        isPending: false,
      };
    case constants.GET_USERS_PENDING:
      return {
        ...state,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.GET_USERS_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: error.message,
      };
    case constants.GET_USERS_BY_ID_FULFILLED:
      return {
        ...state,
        user: payload.data,
        isPending: false,
      };
    case constants.GET_USERS_BY_ID_PENDING:
      return {
        ...state,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.GET_USERS_BY_ID_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: error.message,
      };
    case constants.REMOVE_USER_FULFILLED:
      return {
        ...state,
        user: null,
        isPending: false,
      };
    case constants.REMOVE_USER_PENDING:
      return {
        ...state,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.REMOVE_USER_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: error.message,
      };
    default:
      return state;
  }
};

export default users;
