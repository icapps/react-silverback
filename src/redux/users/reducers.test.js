import users from './reducers';
import constants from './constants';
import defaultInitialState from '../defaultInitialState';

const initialState = {
  userList: [],
  user: null,
  usersCount: 0,
  ...defaultInitialState,
};

describe('users reducer', () => {
  it('should return the initial state', () => {
    expect(users(undefined, {})).toMatchSnapshot();
  });

  it('should GET_USERS_FULFILLED', () => {
    expect(
      users(initialState, {
        type: constants.GET_USERS_FULFILLED,
        payload: { data: {}, meta: { totalCount: 10 } },
      })
    ).toMatchSnapshot();
  });
  it('should GET_USERS_PENDING', () => {
    expect(
      users(initialState, {
        type: constants.GET_USERS_PENDING,
      })
    ).toMatchSnapshot();
  });
  it('should GET_USERS_REJECTED', () => {
    expect(
      users(initialState, {
        type: constants.GET_USERS_REJECTED,
        payload: { errors: [{ detail: 'ErrorMessage' }] },
      })
    ).toMatchSnapshot();
  });

  // GET_USERS_BY_ID
  it('should GET_USERS_BY_ID_FULFILLED', () => {
    expect(
      users(initialState, {
        type: constants.GET_USERS_BY_ID_FULFILLED,
        payload: { data: {}, meta: { totalCount: 10 } },
      })
    ).toMatchSnapshot();
  });
  it('should GET_USERS_BY_ID_PENDING', () => {
    expect(
      users(initialState, {
        type: constants.GET_USERS_BY_ID_PENDING,
      })
    ).toMatchSnapshot();
  });
  it('should GET_USERS_BY_ID_REJECTED', () => {
    expect(
      users(initialState, {
        type: constants.GET_USERS_BY_ID_REJECTED,
        payload: { errors: [{ detail: 'ErrorMessage' }] },
      })
    ).toMatchSnapshot();
  });

  //CREATE_USER
  it('should CREATE_USER_FULFILLED', () => {
    expect(
      users(initialState, {
        type: constants.CREATE_USER_FULFILLED,
        payload: { data: { user: 'test' }, meta: { totalCount: 10 } },
      })
    ).toMatchSnapshot();
  });
  it('should CREATE_USER_PENDING', () => {
    expect(
      users(initialState, {
        type: constants.CREATE_USER_PENDING,
      })
    ).toMatchSnapshot();
  });
  it('should CREATE_USER_REJECTED', () => {
    expect(
      users(initialState, {
        type: constants.CREATE_USER_REJECTED,
        payload: { errors: [{ detail: 'ErrorMessage' }] },
      })
    ).toMatchSnapshot();
  });
  // UPDATE_USER
  it('should UPDATE_USER_FULFILLED', () => {
    expect(
      users(initialState, {
        type: constants.UPDATE_USER_FULFILLED,
        payload: { data: {}, meta: { totalCount: 10 } },
      })
    ).toMatchSnapshot();
  });
  it('should UPDATE_USER_PENDING', () => {
    expect(
      users(initialState, {
        type: constants.UPDATE_USER_PENDING,
      })
    ).toMatchSnapshot();
  });
  it('should UPDATE_USER_REJECTED', () => {
    expect(
      users(initialState, {
        type: constants.UPDATE_USER_REJECTED,
        payload: { errors: [{ detail: 'ErrorMessage' }] },
      })
    ).toMatchSnapshot();
  });

  //REMOVE_USER
  it('should REMOVE_USER_FULFILLED', () => {
    expect(
      users(initialState, {
        type: constants.REMOVE_USER_FULFILLED,
      })
    ).toMatchSnapshot();
  });
  it('should REMOVE_USER_PENDING', () => {
    expect(
      users(initialState, {
        type: constants.REMOVE_USER_PENDING,
      })
    ).toMatchSnapshot();
  });
  it('should REMOVE_USER_REJECTED', () => {
    expect(
      users(initialState, {
        type: constants.REMOVE_USER_REJECTED,
        payload: { errors: [{ detail: 'ErrorMessage' }] },
      })
    ).toMatchSnapshot();
  });
});
