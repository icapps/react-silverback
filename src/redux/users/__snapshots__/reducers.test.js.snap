// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users reducer should CREATE_USER_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isCreateError": false,
  "isCreatePending": false,
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": Object {
    "user": "test",
  },
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should CREATE_USER_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isCreateError": false,
  "isCreatePending": true,
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should CREATE_USER_REJECTED 1`] = `
Object {
  "errorMessage": "Something went wrong!",
  "isCreateError": true,
  "isCreatePending": false,
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should GET_USER_ROLES_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "userRoles": Object {},
  "usersCount": 0,
}
`;

exports[`users reducer should GET_USER_ROLES_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": true,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should GET_USER_ROLES_REJECTED 1`] = `
Object {
  "errorMessage": "Something went wrong!",
  "isError": true,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should GET_USERS_BY_ID_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": Object {},
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should GET_USERS_BY_ID_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isCreateError": false,
  "isError": false,
  "isPending": true,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should GET_USERS_BY_ID_REJECTED 1`] = `
Object {
  "errorMessage": "Something went wrong!",
  "isError": true,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should GET_USERS_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Object {},
  "usersCount": 10,
}
`;

exports[`users reducer should GET_USERS_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isCreateError": false,
  "isError": false,
  "isPending": true,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should GET_USERS_REJECTED 1`] = `
Object {
  "errorMessage": "Something went wrong!",
  "isError": true,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should REMOVE_USER_FULFILLED 1`] = `
Object {
  "deletedUser": null,
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should REMOVE_USER_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": true,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should REMOVE_USER_REJECTED 1`] = `
Object {
  "errorMessage": "Something went wrong!",
  "isError": true,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should RESET_DELETED_USER 1`] = `
Object {
  "deletedUser": "",
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should UPDATE_USER_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": true,
  "user": Object {},
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should UPDATE_USER_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "isUpdatePending": true,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should UPDATE_USER_REJECTED 1`] = `
Object {
  "errorMessage": "Something went wrong!",
  "isError": true,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "user": null,
  "userList": Array [],
  "usersCount": 0,
}
`;

exports[`users reducer should return the initial state 1`] = `
Object {
  "deletedUser": "",
  "errorMessage": "",
  "isCreateError": false,
  "isCreatePending": false,
  "isError": false,
  "isPending": false,
  "isUpdatePending": false,
  "isUserUpdated": false,
  "sortField": "email",
  "sortOrder": "asc",
  "sortValue": "email",
  "user": null,
  "userList": Array [],
  "userRoles": Array [],
  "usersCount": 0,
}
`;
